### PR TITLE
[EXPLORER] Start menu: Fix Small Icons flag

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose
Follow-up to #6646.
Fix wrong flag use: https://github.com/reactos/reactos/pull/6646#issuecomment-2009416127
JIRA issue: [CORE-19494](https://jira.reactos.org/browse/CORE-19494)

## Proposed changes

- Delete `SmallStartMenu` flag (wrong) from `_TW_STUCKRECTS2`.
- Rename `_TW_STUCKRECTS2.SmallIcons` as `SmSmallIcons`.
- Use `SmSmallIcons` flag.

### CODE REVIEW

File: base/shell/explorer/settings.cpp [hunk 1]
```diff
@@ -82,7 +82,6 @@ BOOL TaskbarSettings::Load()
         sr.AlwaysOnTop = TRUE;
         sr.SmallIcons = TRUE;
```
Comment (HBelusca) [inline base/shell/explorer/settings.cpp:83]: "SmSmallIcons", and set it to FALSE by default (i.e. default is large icons)

**Review by JoachimHenze — APPROVED** [review]

File: base/shell/explorer/settings.cpp (hunk 1 above)
Comment (katahiromz) [reply base/shell/explorer/settings.cpp:83]:
```suggestion
        sr.SmallIcons = FALSE;
```

File: base/shell/explorer/precomp.h [hunk 2]
```diff
@@ -205,7 +205,6 @@ typedef struct _TW_STUCKRECTS2
             DWORD AlwaysOnTop : 1;
             DWORD SmallIcons : 1;
```
Comment (HBelusca) [inline base/shell/explorer/precomp.h:206]:
```suggestion
            DWORD SmSmallIcons : 1;
```

File: base/shell/explorer/settings.cpp [hunk 3]
```diff
@@ -80,9 +80,8 @@ BOOL TaskbarSettings::Load()
         sr.Position = ABE_BOTTOM;
         sr.AutoHide = FALSE;
         sr.AlwaysOnTop = TRUE;
-        sr.SmallIcons = TRUE;
+        sr.SmallIcons = FALSE;
```
Comment (HBelusca) [inline base/shell/explorer/settings.cpp:83]:
```suggestion
        sr.SmSmallIcons = FALSE;
```

File: base/shell/explorer/startmnucust.cpp [hunk 4]
```diff
@@ -91,12 +91,12 @@ static VOID CALLBACK CustomSetAdvanced(const CUSTOM_ENTRY *entry, BOOL bValue)
 
 static BOOL CALLBACK CustomGetSmallStartMenu(const CUSTOM_ENTRY *entry)
 {
-    return g_TaskbarSettings.sr.SmallStartMenu;
+    return g_TaskbarSettings.sr.SmallIcons;
```
Comment (HBelusca) [inline base/shell/explorer/startmnucust.cpp:94]:
```suggestion
    return g_TaskbarSettings.sr.SmSmallIcons;
```

File: base/shell/explorer/startmnucust.cpp [hunk 5]
```diff
@@ -91,12 +91,12 @@ static VOID CALLBACK CustomSetAdvanced(const CUSTOM_ENTRY *entry, BOOL bValue)
...
 }
 
 static VOID CALLBACK CustomSetSmallStartMenu(const CUSTOM_ENTRY *entry, BOOL bValue)
 {
-    g_TaskbarSettings.sr.SmallStartMenu = bValue;
+    g_TaskbarSettings.sr.SmallIcons = bValue;
```
Comment (HBelusca) [inline base/shell/explorer/startmnucust.cpp:99]:
```suggestion
    g_TaskbarSettings.sr.SmSmallIcons = bValue;
```

File: base/shell/explorer/traywnd.cpp [hunk 6]
```diff
@@ -2554,9 +2554,9 @@ class CTrayWindow :
         RegLoadSettings();
 
         /* Create and initialize the start menu */
-        BOOL bSmallStartMenu = g_TaskbarSettings.sr.SmallStartMenu;
         HBITMAP hbmBanner = LoadBitmapW(hExplorerInstance, MAKEINTRESOURCEW(IDB_STARTMENU));
-        m_StartMenuPopup = CreateStartMenu(this, &m_StartMenuBand, hbmBanner, bSmallStartMenu);
+        m_StartMenuPopup = CreateStartMenu(this, &m_StartMenuBand, hbmBanner,
+                                           g_TaskbarSettings.sr.SmallIcons);
```
Comment (HBelusca) [inline base/shell/explorer/traywnd.cpp:2559]:
```suggestion
                                           g_TaskbarSettings.sr.SmSmallIcons);
```

File: base/shell/explorer/traywnd.cpp [hunk 7]
```diff
@@ -2665,9 +2665,9 @@ class CTrayWindow :
             HideStartMenu();
             m_StartMenuBand.Release();
 
-            BOOL bSmallStartMenu = g_TaskbarSettings.sr.SmallStartMenu;
             HBITMAP hbmBanner = LoadBitmapW(hExplorerInstance, MAKEINTRESOURCEW(IDB_STARTMENU));
-            m_StartMenuPopup = CreateStartMenu(this, &m_StartMenuBand, hbmBanner, bSmallStartMenu);
+            m_StartMenuPopup = CreateStartMenu(this, &m_StartMenuBand, hbmBanner,
+                                               g_TaskbarSettings.sr.SmallIcons);
```
Comment (HBelusca) [inline base/shell/explorer/traywnd.cpp:2670]:
```suggestion
                                               g_TaskbarSettings.sr.SmSmallIcons);
```

**Review by HBelusca — COMMENTED** [review]
The changes for "SmallIcons" --> "SmSmallIcons" (cf. my previous comment).

File: base/shell/explorer/traywnd.cpp [hunk 8]
```diff
@@ -2665,7 +2665,7 @@ class CTrayWindow :
             HideStartMenu();
             m_StartMenuBand.Release();
 
-            BOOL bSmallStartMenu = g_TaskbarSettings.sr.SmallStartMenu;
+            BOOL bSmallStartMenu = g_TaskbarSettings.sr.SmallIcons;
             HBITMAP hbmBanner = LoadBitmapW(hExplorerInstance, MAKEINTRESOURCEW(IDB_STARTMENU));
             m_StartMenuPopup = CreateStartMenu(this, &m_StartMenuBand, hbmBanner, bSmallStartMenu);
```
[the remark this answers is not in the document]
Comment (katahiromz) [reply base/shell/explorer/traywnd.cpp]: image
Not working well.

File: base/shell/explorer/precomp.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (katahiromz) [reply base/shell/explorer/precomp.h]: Renamed.

File: base/shell/explorer/traywnd.cpp (hunk 8 above)
[the remark this answers is not in the document]
Comment (HBelusca) [reply base/shell/explorer/traywnd.cpp]: OK then let's keep the start menu recreation code as it was originally. The update seems to just reduce the size of the icons/remove/add banner only, while the size of the menu itself stays the same.

File: base/shell/explorer/CMakeLists.txt [hunk 9]
```diff
@@ -13,7 +13,6 @@ list(APPEND SOURCE
     settings.cpp
     shellservice.cpp
     startctxmnu.cpp
-    startmnu.cpp
```
Comment (HBelusca) [inline base/shell/explorer/CMakeLists.txt:16]: I would suggest you revert this change -- as well as the change of the CreateStartMenu() function -- to keep the PR simple enough. (Also there is no reason why to move the code of this file to somewhere else. You've put the start menu creation code into tray window file, which doesn't make sense from the point of view of code separation)

**Review by HBelusca — CHANGES_REQUESTED** [review]

File: base/shell/explorer/CMakeLists.txt (hunk 9 above)
Comment (katahiromz) [reply base/shell/explorer/CMakeLists.txt]: Reverted.

**Comment by HBelusca** [comment]
OK with the revert :+1:
